File: 2000/2011.cpp
#include<iostream>
using namespace std;
#define ll long long

string s;
int num[5001];
ll dp[5001]{ 1,1 };

void input() {
	cin >> s;
	for (int i = 0; i < s.length(); ++i) {
		num[i] = s[i] - '0';
	}
}

void solution() {
	input();
	if (num[0] == 0) {
		cout << 0;
		return;
	}
	dp[0] = dp[1] = 1;
	for (int i = 2; i <= s.length(); ++i) {
		int n = num[i - 2] * 10 + num[i - 1];
		if(num[i - 1] > 0) dp[i] = dp[i - 1];
		if (n <= 26 && n >= 10) {
			dp[i] = (dp[i] + dp[i - 2])% 1000000;
		}
	}
	cout << dp[s.length()];
}

int main() {
	cin.tie(0), cout.tie(0), ios_base::sync_with_stdio(0);
	solution();
	return 0;
}

File: 2000/2011.java
import java.io.*;
import java.util.*;

public class Main {

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static String s;
	static int n;
	static int[] dp;
	
	
    static void input() throws IOException {
    	s = br.readLine();
    	n = s.length();
    	dp = new int[n + 1];
    }
    
    static void solution() {
   		if(s.charAt(0) == '0') {
   			System.out.println(0);
   			return;
   		}
    	dp[0] = dp[1] = 1;
    	for(int i = 2;i <= n; ++i) {
    		int num = Integer.parseInt(s.substring(i - 2, i));
    		if(num % 10 == 0) {
    			if(num == 10 || num == 20) {
    				dp[i] = dp[i - 2] % (int)1e6;
    			} else break;
    		} else if(num >= 10 && num <= 26) {
    			dp[i] = (dp[i - 1] + dp[i - 2]) % (int)1e6;
    		} else {
    			dp[i] = dp[i - 1] % (int)1e6;
    		}
    	}
    	System.out.println(dp[n] % (int)1e6);
    }
    
    public static void main(String[] args) throws IOException {
   		input();
   		solution();
    }
}
